src/main/java/namor/template/main/App.java

```diff
@@ -1,9 +1,64 @@
 package namor.template.main;
 
+import java.util.Arrays;
+
 // TODO: 20.03.17 [GUSb] A test class to check everything works, should be deleted with first finished task
 public class App {
 
-    public int add(int a, int b) {
-        return a + b;
+    private int top;
+    int size;
+    int[] stack;
+
+    public App(int arraySize)
+    {
+        size = arraySize;
+        stack = new int[size];
+        top = 0;
     }
+
+    public boolean IsEmpty()
+    {
+        return top == 0;
+    }
+
+    public void push(int value)
+    {
+        if (top == stack.length)
+        {
+            extendStack();
+        }
+        stack[top] = value;
+        top++;
+    }
+
+    public int stackSize()
+    {
+        return top;
+    }
+
+    public void pop()
+    {
+        if(!IsEmpty())
+        {
+            stack[top-1] = 0;
+            top--;
+        }
+
+    }
+
+    public int peak()
+    {
+        if (!IsEmpty())
+        {
+            return stack[top-1];
+        }
+        else return -1;
```

Comment:
In case of situations when you see that sb can make some action which can cause a bug or crash of app you can use Exceptions. So when there is no values in stack but sb try to pop the value **throw the exception** with message that tells what's went wrong